pkg/controller/master/upgrade/upgrade_controller.go

```diff
@@ -864,3 +873,44 @@ func (h *upgradeHandler) checkLogReadyCondition(upgrade *harvesterv1.Upgrade) (*
 	h.upgradeController.EnqueueAfter(upgrade.Namespace, upgrade.Name, time.Second*5)
 	return upgrade, nil
 }
+
+func (h *upgradeHandler) disableAddons(upgrade *harvesterv1.Upgrade) error {
+	addon, err := h.addonCache.Get(util.KubeSystemNamespace, util.DeschedulerAddon)
+	if err != nil && !apierrors.IsNotFound(err) {
+		return err
+	}
+
+	if addon != nil && addon.Spec.Enabled {
+		logrus.Info("Descheduler addon is enabled, disabling it for upgrade")
+		upgrade.Annotations[util.AnnotationReenableDeschedulerAddon] = "true"
```

Comment:
the annotation `AnnotationReenableDeschedulerAddon` seems need to be put on `addon` object with the value of upgrade name/UID, at least more than `upgrade` object, as there is case:

`addon object is disabled successfully, then upgrade object is failed to update`, and when reconciller comes again, upgrade controller only finds the addon is disabled

updating 2+ objects on one controller logic, will face such races

Reply:
even with the locker, it still does not mean both CRD object updating are always successful in the mean time